source/slang/slang-ir-autodiff-transpose.h

```diff
@@ -2429,25 +2455,38 @@ struct DiffTransposePass
         auto baseType = firstFwdSwizzleInst->getBase()->getDataType();
 
         IRIntegerValue elementCount = 0;
-        IRType* elementType = nullptr;
-        IRType* primalElementType = nullptr;
+        List<IRType*> elementTypes;
+        List<IRType*> primalElementTypes;
         bool isVectorType = false;
-
+        bool isTupleType = false;
         if (auto vectorType = as<IRVectorType>(baseType))
         {
             IRInst* elementCountInst = vectorType->getElementCount();
-            elementType = vectorType->getElementType();
-            primalElementType = as<IRVectorType>(aggPrimalType)->getElementType();
-            SLANG_ASSERT(as<IRIntLit>(elementCountInst));
             elementCount = as<IRIntLit>(elementCountInst)->getValue();
+            for (IRIntegerValue i = 0; i < elementCount; i++)
+            {
+                elementTypes.add(vectorType->getElementType());
+                primalElementTypes.add(as<IRVectorType>(aggPrimalType)->getElementType());
+            }
```

Comment:
We may want to reserve the space for `elementTypes` with `elementCount`.
It will be a good idea to prepare for an extreme user shader that may use many elements in a tuple.